src/java/org/apache/cassandra/locator/ReplicaPlans.java

```diff
@@ -93,6 +96,8 @@ public class ReplicaPlans
 
     private static final Range<Token> FULL_TOKEN_RANGE = new Range<>(DatabaseDescriptor.getPartitioner().getMinimumToken(), DatabaseDescriptor.getPartitioner().getMinimumToken());
 
+    private static final int REQUIRED_BATCHLOG_REPLICA_COUNT = Math.min(2, CassandraRelevantProperties.REQUIRED_BATCHLOG_REPLICA_COUNT.getInt());
```

Comment:
nit:  this is going to be confusing for the user, they set a higher value, haven't read the docs properly, and are not getting any feedback we're capping the value at `2`.

```suggestion
    private static final int REQUIRED_BATCHLOG_REPLICA_COUNT = Math.min(2, CassandraRelevantProperties.REQUIRED_BATCHLOG_REPLICA_COUNT.getInt());
    
    static
    {
        int batchlogReplicaCount = CassandraRelevantProperties.REQUIRED_BATCHLOG_REPLICA_COUNT.getInt();
        Preconditions.checkArgument(1 <= batchlogReplicaCount && batchlogReplicaCount <= 2, "%s only accepts values of 1 or 2.",  CassandraRelevantProperties.REQUIRED_BATCHLOG_REPLICA_COUNT.getKey());

        logger.warn("System property {} was set to {} but is capped at 2." CassandraRelevantProperties.REQUIRED_BATCHLOG_REPLICA_COUNT.getKey(), batchlogReplicaCount);
    }
```

🤷 


(IIUC you can't set the system property to zero …?)

Reply:
i'm thinking this is better
```
private static final int REQUIRED_BATCHLOG_REPLICA_COUNT = Math.max(1, Math.min(2, CassandraRelevantProperties.REQUIRED_BATCHLOG_REPLICA_COUNT.getInt()));

int batchlogReplicaCount = CassandraRelevantProperties.REQUIRED_BATCHLOG_REPLICA_COUNT.getInt();
if (1 <= batchlogReplicaCount && batchlogReplicaCount <= 2)
    logger.warn("System property {} was set to {} but must be 1 or 2. Running with {}" CassandraRelevantProperties.REQUIRED_BATCHLOG_REPLICA_COUNT.getKey(), batchlogReplicaCount, REQUIRED_BATCHLOG_REPLICA_COUNT);
```

Reply:
see my last suggestion :-)